File: app/Module.java
import com.google.inject.AbstractModule;
import org.pcap4j.core.PcapNativeException;
import org.pcap4j.core.PcapNetworkInterface;
import org.pcap4j.core.Pcaps;
import play.Logger;
import play.libs.akka.AkkaGuiceSupport;

/**
 * Created by Kristian Lange on 2017.
 */
public class Module extends AbstractModule implements AkkaGuiceSupport {

    private Logger.ALogger logger = Logger.of(Module.class);

    @Override
    protected void configure() {
        // Print out all network interfaces so users know which names they have
        try {
            for (PcapNetworkInterface nif : Pcaps.findAllDevs()) {
                Logger.info("NIF: " + nif.toString());
            }
        } catch (PcapNativeException e) {
            // do nothing
        }
    }
}

File: app/Module.scala
import com.google.inject.AbstractModule
import org.pcap4j.core.{PcapNativeException, Pcaps}
import play.api.{Configuration, Environment, Logger}

import scala.collection.JavaConverters._

class Module(environment: Environment, configuration: Configuration) extends AbstractModule {

  private val logger: Logger = Logger(this.getClass)

  def configure() = {
    // Print out all network interfaces so users know which names they have
    try {
      val nifs = Pcaps.findAllDevs.asScala

      nifs.foreach(nif => Logger.info("NIF: " + nif.toString))

      System.out.println("List of available network interfaces:")
      nifs.foreach(nif => System.out.println(s"  ${nif.getName}"))

      var httpAddress = configuration.get[String]("play.server.http.address")
      if (httpAddress == "0.0.0.0") httpAddress = "localhost"
      val httpPort = configuration.get[Int]("play.server.http.port")
      val firstNif = if (nifs.nonEmpty) nifs.head.getName else "myNif"
      System.out.println(s"E.g. in your browser use URL " +
          s"$httpAddress:$httpPort/glimpse?nif=$firstNif")
    } catch {
      case e: PcapNativeException =>
      // do nothing
    }
  }
}
